tests/output.rs

```diff
@@ -0,0 +1,174 @@
+    pub lines: Vec<String>,
+    pub range: Range,
+}
+
+impl ExpectedMatch {
+    pub fn expected_output_text(&self) -> String {
+        self.lines
+            .iter()
+            .enumerate()
+            .map(|(line_index, line)| {
+                format!(
+                    "{}:{}:{}",
+                    self.relative_file_path,
+                    self.range.start.line + line_index + 1,
+                    line
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("\n")
+    }
+}
+
+const MATCH_OUTPUT_LINE_REGEX_STR: &'static str = r#"(^|\n).+:\d+:"#;
+
+fn predicate_from_expected_matches(expected_matches: &[ExpectedMatch]) -> BoxPredicate<str> {
+    expected_matches.into_iter().fold(
+        BoxPredicate::new(
+            predicate::str::is_match(MATCH_OUTPUT_LINE_REGEX_STR)
+                .unwrap()
+                .count(
+                    expected_matches
+                        .into_iter()
+                        .map(|expected_match| expected_match.lines.len())
+                        .sum(),
+                ),
+        ),
+        |predicate, expected_match| {
+            BoxPredicate::new(predicate.and(predicate_from_expected_match(expected_match)))
+        },
+    )
+}
+
+fn predicate_from_expected_match(expected_match: &ExpectedMatch) -> BoxPredicate<str> {
+    BoxPredicate::new(predicate::str::contains(
+        expected_match.expected_output_text(),
+    ))
+}
+
+fn assert_expected_matches<TArg: AsRef<OsStr>>(
+    args: impl IntoIterator<Item = TArg>,
+    current_dir: impl AsRef<Path>,
+    expected_matches: &[ExpectedMatch],
+) {
+    Command::cargo_bin("tree-sitter-grep")
+        .unwrap()
+        .args(args)
+        .current_dir(current_dir)
+        .assert()
+        .success()
+        .stdout(predicate_from_expected_matches(expected_matches));
+}
+
+struct FixtureQueryExpectedMatches {
+    pub fixture_dir_name: String,
+    pub language: String,
+    pub query_source: String,
+    pub expected_matches: Vec<ExpectedMatch>,
+}
+
+impl FixtureQueryExpectedMatches {
+    pub fn assert_expected_matches(&self) {
+        assert_expected_matches(
+            [
+                "--query-source",
+                &self.query_source,
+                "--language",
+                &self.language,
+            ],
+            get_fixture_dir_path_from_name(&self.fixture_dir_name),
+            &self.expected_matches,
+        );
+    }
+}
+
+fn get_fixture_dir_path_from_name(fixture_dir_name: &str) -> PathBuf {
+    // per https://andrewra.dev/2019/03/01/testing-in-rust-temporary-files/
+    let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let mut path: PathBuf = root_dir.into();
+    path.push("tests/fixtures");
+    path.push(fixture_dir_name);
+    path
+}
+
+const FUNCTION_ITEM_QUERY_SOURCE: &'static str = "(function_item) @function_item";
+
+static RUST_PROJECT_FUNCTION_ITEM_EXPECTED_MATCHES: Lazy<FixtureQueryExpectedMatches> =
```

Comment:
I guess I'm picturing these being reusable for testing eg different output formats?